lemur/models.py

```diff
@@ -0,0 +1,272 @@
+        return self.password == password
+
+    # can function checks if user is allowed to peform an operation
+    def can(self, permissions):
+        return self.role is not None and (
+            self.role.permissions & permissions) == permissions
+
+    def __repr__(self):
+        tpl = ('User<id: {id}, username: {username}, password: {password},'
+               ' role: {role_name}, classes: {classes}, labs: {labs}>')
+        formatted = tpl.format(id=self.id, username=self.username,
+                               password=self.password,
+                               role_name=self.role_name,
+                               classes=[c.id for c in self.classes],
+                               labs=[l.id for l in self.labs])
+        return formatted
+
+
+# Function for use when user is not logged in
+class AnonymousUser(AnonymousUserMixin):
+    def can(self, permissions):
+        return False
+# Associate the LoginManager with the anonymous user class
+login_manager.anonymous_user = AnonymousUser
+
+
+# Function accepts a user id and returns the user object of that id
+@login_manager.user_loader
+def load_user(user_id):
+    return User.query.get(user_id)
+
+
+class Role(db.Model):
+    __tablename__ = 'Role'
+    name = db.Column(db.String(64), nullable=False, unique=True,
+                     primary_key=True)
+
+    # A number representing the power the role has using bit operation
+    permissions = db.Column(db.Integer, nullable=False)
+
+    # one-to-many: A Role can have multiple users
+    users = db.relationship('User', back_populates='role', lazy='dynamic')
+
+    # For database initialization, no object needed to use this method
+    @staticmethod
+    def insert_roles():
+        roles = {
+            'Student': (Permission.DATA_ENTRY),
+            'Admin': (Permission.DATA_ENTRY |
+                      Permission.DATA_EDIT |
+                      Permission.LAB_SETUP |
+                      Permission.ADMIN),
+            'SuperAdmin': (Permission.DATA_ENTRY |
+                           Permission.DATA_EDIT |
+                           Permission.LAB_SETUP |
+                           Permission.ADMIN |
+                           Permission.LAB_MANAGE |
+                           Permission.USER_MANAGE |
+                           Permission.SUPERADMIN)
+        }
+        for r in roles:
+            role = Role.query.filter_by(name=r).first()
+            if role is None:
+                role = Role(name=r)
+            role.permissions = roles[r]
+            db.session.add(role)
+        db.session.commit()
+
+    def __repr__(self):
+        tpl = ('Role<name: {name},'
+               ' permissions: {permissions}>')
+        formatted = tpl.format(name=self.name,
+                               permissions=self.permissions)
+        return formatted
+
+
+# Bits associated with the role power they represent using hexadecimal format
+class Permission:
+    DATA_ENTRY = 0x01
```

Comment:
If I change it to a string then I assume that I am supposed to store the permission variable as an array of string? Each time when the app checks the permission of the current user, it then looks up the permissions array associated the user's role. Is this the design you suggest?